packages/aws-cdk/lib/commands/flag-operations.ts

```diff
@@ -40,477 +47,684 @@ interface FlagOperationsParams {
 
   /** User ran --unconfigured option */
   unconfigured?: boolean;
-}
 
-export async function handleFlags(flagData: FeatureFlag[], ioHelper: IoHelper, options: FlagsOptions, toolkit: Toolkit) {
-  flagData = flagData.filter(flag => !OBSOLETE_FLAGS.includes(flag.name));
+  /** User ran --safe option */
+  safe?: boolean;
+
+  /** User provided the number of threads to run the --safe operation with
+   * @default 4
+   */
+  concurrency?: number;
+
+  /** User provided --app option */
+  app?: string;
+
+  /** User ran --interactive option */
+  interactive?: boolean;
 
-  if (flagData.length == 0) {
-    await ioHelper.defaults.error('The \'cdk flags\' command is not compatible with the AWS CDK library used by your application. Please upgrade to 2.212.0 or above.');
-    return;
+  /** User ran --set option */
+  set?: boolean;
+}
+
+export class DetermineSafeFlags {
+  private readonly params: FlagOperationsParams;
+
+  constructor(flagData: FeatureFlag[], ioHelper: IoHelper, options: HandleFlagsOptions, toolkit: Toolkit) {
+    this.params = {
+      flagData: flagData.filter(flag => !OBSOLETE_FLAGS.includes(flag.name)),
+      toolkit,
+      ioHelper,
+      recommended: options.recommended,
```

Comment:
Couldn't the members of this class be:

```ts
private readonly flags: FeatureFlag[];
private readonly toolkit: Toolkit;
private readonly ioHelper: IoHelper;
private readonly options: Options;
```

Rather than combine everything into a single struct?